pkg/objstore/prefixed_bucket_test.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package objstore
+
+import (
+	"context"
+	"io/ioutil"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/thanos-io/thanos/pkg/testutil"
+)
+
+func TestPrefixedBucket_Acceptance(t *testing.T) {
+	prefix := "/someprefix/anotherprefix/"
```

Comment:
Since I believe that you will clean up the code a bit either way, could we put these strings into a string slice and iterate over the elements, and call those functions? I think it would make this way neater.